src/permits/permits.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { PermitCode } from './permit.service';
+import * as mongoose from 'mongoose';
+
+export const PERMIT_MODEL_NAME = 'Permit';
+
+export interface PermitModel extends PermitCode, mongoose.Document {}
+
+export const permitSchema = new mongoose.Schema(
+    { code: String, reasonAbbrevation: String },
```

Comment:
```suggestion
    { 
        code: { type: String, unique: true }, 
        reasonAbbrevation: String 
    },
```

dann kümmert sich Mongoose um Unique Entries.